ibis/expr/types.py

```diff
@@ -1016,6 +1016,31 @@ def _implicit_cast(self, arg):
         return DateScalar(op)
 
 
+class TimeValue(TemporalValue):
+
+    def type(self):
+        return dt.time
+
+    def _can_implicit_cast(self, arg):
+        op = arg.op()
+        if isinstance(op, Literal):
+            try:
+                from pandas.core.tools.datetimes import to_time
+                to_time(op.value)
+                return True
+            except ValueError:
+                return False
+        return False
+
+    def _can_compare(self, other):
+        return isinstance(other, (TemporalValue, StringValue))
```

Comment:
Here are the postgres rules:

1. `timestamp` compare `date`: allowed
2. `timestamp` compare `time`: not allowed
3. `date` compare `time`: not allowed

So, you can pretty much only compare `time`s with other `time` typed values. I think that's a reasonable default for now.